signer/migrations/0003__create_tables.sql

```diff
@@ -184,3 +191,111 @@ CREATE TABLE sbtc_signer.withdrawal_reject_events (
+-- which can be retrieved from the `withdrawal_requests` table.
+CREATE TABLE sbtc_signer.swept_withdrawals (
+    -- Internal ID of the swept withdrawal.
+    id BIGSERIAL PRIMARY KEY,
+    -- References the `packaged_transaction` in which this withdrawal was
+    -- included.
+    sweep_transaction_id INTEGER NOT NULL,
+    -- The index of the sweep output in the packaged transaction.
+    output_index INTEGER NOT NULL,
+    -- The ID of the withdrawal request, referencing the `withdrawal_requests`
+    -- table.
+    withdrawal_request_id BIGINT NOT NULL,
+    -- The Stacks block hash of the withdrawal request, referencing the
+    -- `withdrawal_requests` table.
+    withdrawal_request_block_hash BYTEA NOT NULL,
+
+    FOREIGN KEY (sweep_transaction_id) 
+        REFERENCES sbtc_signer.sweep_transactions(id),
+
+    FOREIGN KEY (withdrawal_request_id, withdrawal_request_block_hash) 
+        REFERENCES sbtc_signer.withdrawal_requests(request_id, block_hash)
+);
+-- Our main index which will cover searches by 'withdrawal_request_id' and
+-- 'withdrawal_request_block_hash' while also restricting the combination to be
+-- unique per 'sweep_transaction_id'.
+CREATE UNIQUE INDEX uix_swept_req_id_req_block_hash_pkgd_txid 
+    ON sbtc_signer.swept_withdrawals(withdrawal_request_id, withdrawal_request_block_hash, sweep_transaction_id);
+
+-- Represents a single deposit request which has been included in a
+-- transaction package. Deposit requests do not have a unique ID in the same way
+-- as withdrawal requests, so we reference the Bitcoin transaction ID and output
+-- index instead, which can be retrieved from the `deposit_requests` table.
+CREATE TABLE sbtc_signer.swept_deposits (
+    -- Internal ID of the swept deposit.
+    id BIGSERIAL PRIMARY KEY,
+    -- References the `packaged_transaction` in which this deposit was included.
+    sweep_transaction_id INTEGER NOT NULL,
+    -- The index of the sweep output in the packaged transaction.
+    output_index INTEGER NOT NULL,
+    -- The Bitcoin transaction ID of the deposit request, referencing the
+    -- `deposit_requests` table.
+    deposit_request_txid BYTEA NOT NULL,
+    -- The output index of the deposit request, referencing the
+    -- `deposit_requests` table.
+    deposit_request_output_index INTEGER NOT NULL,
+
+    FOREIGN KEY (sweep_transaction_id)
+        REFERENCES sbtc_signer.sweep_transactions(id),
+
+    FOREIGN KEY (deposit_request_txid, deposit_request_output_index) 
+        REFERENCES sbtc_signer.deposit_requests(txid, output_index)
+);
+-- Our main index which will cover searches by 'deposit_request_txid' and
+-- 'deposit_request_output_index', while also restricting the combination to be
+-- unique per 'sweep_transaction_id'.
+CREATE UNIQUE INDEX uix_swept_deposits_req_txid_req_output_index_pkgd_txid
+    ON sbtc_signer.swept_deposits(deposit_request_txid, deposit_request_output_index, sweep_transaction_id);
```

Comment:
Hmmm, although I like all of my tables to have an `id` column, I can also see the case for removing them when we have a natural composite primary. In this case, it seems like we have a natural primary key using the columns in this index.

Reply:
I generally avoid natural keys, even composite, because they have almost always bitten me in the a** when requirements change down the line. A surrogate key I can **guarantee** will always uniquely identify the record, even if the natural key were to change (i.e. gain or lose a column), and also _allows_ it.

I've got years of experience of natural keys biting me in the a**, but blockchain is a new domain for me and I'll admit that, while my defaults kick in to not trust them, it may not be the right choice in this domain. Let's huddle it (with others who maybe want to discuss) when we get the chance?

Reply:
Yeah let's do that. I'll set something up for Monday.

Reply:
From our DB meeting, I will revert to natural keys.

Reply:
Natural keys implemented